airtable/params.py

```diff
@@ -1,368 +1,462 @@
+    or
+    >>> params={'sort': [('FieldOne', 'asc'), ('-FieldTwo', 'desc')]}
+
+    Requests Url Params Encoding:
+    >>> ?sort=field&sort=direction&sort=field&sort=direction
+
+    Expected Url Params:
+    >>> ?sort[0][field]=FieldOne&sort[0][direction]=asc
+    """
+
+    def to_param_dict(self):
+        """ Sorts to ensure Order is consistent for Testing """
+        param_dict = {}
+        for index, dictionary in enumerate(self.value):
+            for key, value in dictionary.items():
+                param_name = "{param_name}[{index}][{key}]".format(
+                    param_name=self.param_name, index=index, key=key
+                )
+                param_dict[param_name] = value
+        return OrderedDict(sorted(param_dict.items()))
+
+
+class AirtableParams(object):
+    class MaxRecordsParam(_BaseParam):
+        """
+        Max Records Param
+
+        Kwargs:
+            ``max_records=`` or ``maxRecords=``
+
+        The maximum total number of records that will be returned.
+
+        Usage:
+
+        >>> airtable.get_all(max_records=10)
+
+        Args:
+            max_records (``int``): The maximum total number of records that
+                will be returned.
+
+
+        """
+
+        # Class Input > Output
+        # >>> filter = MaxRecordsParam(100)
+        # >>> filter.to_param_dict()
+        # {'maxRecords: 100}
+
+        param_name = "maxRecords"
+        kwarg = "max_records"
+
+    class ViewParam(_BaseParam):
+        """
+        View Param
+
+        Kwargs:
+            ``view=``
+
+        If set, only the records in that view will be returned.
+        The records will be sorted according to the order of the view.
+
+        Usage:
+
+        >>> airtable.get_all(view='My View')
+
+        Args:
+            view (``str``): The name or ID of a view.
+
+        """
+
+        # Class Input > Output
+        # >>> filter = ViewParam('Name or Id Of View')
+        # >>> filter.to_param_dict()
+        # {'view: 'Name or Id Of View'}
+
+        param_name = "view"
+        kwarg = param_name
+
+    class PageSizeParam(_BaseParam):
+        """
+        Page Size Param
+
+        Kwargs:
+            ``page_size=`` or ``pageSize=``
+
+        Limits the maximum number of records returned in each request.
+        Default is 100.
+
+        Usage:
+
+        >>> airtable.get_all(page_size=50)
+
+        Args:
+            page_size (``int``): The number of records returned in each request.
+                Must be less than or equal to 100. Default is 100.
+
+        """
+
+        # Class Input > Output
+        # >>> filter = PageSizeParam(50)
+        # >>> filter.to_param_dict()
+        # {'pageSize: 50}
+
+        param_name = "pageSize"
+        kwarg = "page_size"
+
+    class CellFormatParam(_BaseParam):
+        """
+        cellFormat Param
+
+        Kwargs:
+            ``cellFormat=``
+
+        If set, the format that should be used for cell values. Supported values are:
```

Comment:
seems like supported value are only `string` and `json`, correct?
maybe we should check/validate input to allowed values?